packages/utils/src/features/safe-shield/types.ts

```diff
@@ -89,16 +89,48 @@ export type AnyStatus = RecipientStatus | BridgeStatus | ContractStatus | Threat
 
 export type AnalysisResult<T extends AnyStatus> = { severity: Severity; type: T; title: string; description: string }
 
+export type MasterCopyChangeThreatAnalysisResult = AnalysisResult<ThreatStatus.MASTER_COPY_CHANGE> & {
+  /** Address of the old master copy/implementation contract */
+  before: string
+  /** Address of the new master copy/implementation contract */
+  after: string
+}
+
+export type MaliciousOrModerateThreatAnalysisResult = AnalysisResult<ThreatStatus.MALICIOUS | ThreatStatus.MODERATE> & {
+  /** A potential map of specific issues identified during threat analysis, grouped by severity */
+  issues?: Map<keyof typeof Severity, Array<string>>
+}
+
 export type AddressAnalysisResults = {
-  [_group in StatusGroup]?: AnalysisResult<RecipientStatus | BridgeStatus | ContractStatus>[]
+  [_group in StatusGroup]?: (
+    | AnalysisResult<RecipientStatus | BridgeStatus | ContractStatus | ThreatStatus>
+    | MaliciousOrModerateThreatAnalysisResult
+    | MasterCopyChangeThreatAnalysisResult
+  )[]
 }
 
 export type RecipientAnalysisResults = { [address: string]: AddressAnalysisResults }
 export type ContractAnalysisResults = { [address: string]: AddressAnalysisResults }
-export type ThreatAnalysisResult = AnalysisResult<ThreatStatus>
+export type ThreatAnalysisResult = { [address: string]: AddressAnalysisResults }
```

Comment:
There was a typo already. 
```suggestion
export type ThreatAnalysisResults = { [address: string]: AddressAnalysisResults }
```

Like that we can have a separate type `ThreatAnalysisResult` for only a single result.